lib/hanami/action.rb

```diff
@@ -171,16 +171,25 @@ class Action
     # @since 0.1.0
     # @api private
     def self.inherited(base)
-      base.class_eval do
-        include Utils::ClassAttribute
-        class_attribute :before_callbacks
-        self.before_callbacks = Utils::Callbacks::Chain.new
+      if base.superclass == Hanami::Action
+        base.class_eval do
+          include Utils::ClassAttribute
 
-        class_attribute :after_callbacks
-        self.after_callbacks = Utils::Callbacks::Chain.new
+          class_attribute :before_callbacks
+          self.before_callbacks = Utils::Callbacks::Chain.new
 
-        prepend InstanceMethods
-        include Validatable if defined?(Validatable)
+          class_attribute :after_callbacks
+          self.after_callbacks = Utils::Callbacks::Chain.new
+
+          prepend InstanceMethods
+          include Validatable if defined?(Validatable)
+        end
+      else
+        base.class_eval do
+          self.before_callbacks = superclass.before_callbacks.dup
+          self.after_callbacks  = superclass.after_callbacks.dup
```

Comment:
I wonder if we could make class attributes _always_ duplicate on inherited rather than having to manually arrange something like this in every class that uses them?

Reply:
Good point 👍I checked and this should happen by default: https://github.com/hanami/utils/blob/master/lib/hanami/utils/class_attribute.rb#L80

Let me investigate to see why it doesn't happen here..

Reply:
Actually, I pointed `master`, but should've been use this link https://github.com/hanami/utils/blob/unstable/lib/hanami/utils/class_attribute.rb#L94

Reply:
I turned out that these two lines were redundant. Thanks again @timriley! See https://github.com/hanami/controller/pull/277/commits/d5a829ee75345e9960522484a30977fe4342b5b3

Reply:
Great!